source/features/minimize-upload-bar.tsx

```diff
@@ -0,0 +1,39 @@
+import './minimize-upload-bar.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+
+async function addButton(): Promise<void> {
+	for (const toolbar of select.all('form:not(.rgh-has-upload-field) markdown-toolbar')) {
+		toolbar.lastElementChild!.prepend(
+			<button type="button" className="toolbar-item tooltipped tooltipped-n rgh-upload-btn" aria-label="Attach files">
+				{icons.cloudUpload()}
+			</button>
+		);
+		toolbar.closest('form')!.classList.add('rgh-has-upload-field');
```

Comment:
Any simplifications possible in addButton? e.g. to avoid 1 DOM traversing and skip a part of the selector

Reply:
I thought about it, but the DOM is roughly as following:
* `form`
  * `.previewable-comment-form`
    * `.comment-form-head`
      * <mark>**`mardown-toolbar`**</mark>
    * `file-attachment`
      * `textarea`
      * **`.drag-and-drop`**

Because the toolbar and drag-and-drop message are on different DOM levels, it needs another parent that contains both elements, to get the CSS working. Using `form` element to add the class still seems the best solution.

If you have another suggestion...

Reply:
I meant: `select.all('markdown-toolbar:not(.rgh-has-upload-field) > :last-child')`. This avoid `form` in the selector and `lastElementChild` on the next line

Reply:
For that to work, I needed to add another class to the `markdown-toolbar` to not conflict with the class that is needed for the CSS.